### PR TITLE
Step 6 delete

### PR DESCRIPTION
**Pull request by Obinnabii**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for deleting tasks. */
```
Comment (billyeh) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java]:
```suggestion
/** Servlet responsible for deleting comments. */
```

File: portfolio/src/main/webapp/comments.js [hunk 2]
```diff
@@ -46,12 +48,23 @@ async function getCommentsList() {
...
+function addToList(commentObj, list) {
+  const comment = document.createElement('li');
+  comment.innerText = commentObj.text;
+
+  const deleteButtonElement = document.createElement('button');
+  deleteButtonElement.innerText = 'Delete';
+  deleteButtonElement.addEventListener('click', () => {
+    deleteCommentFromDB(commentObj);
+    comment.remove();
+    getCommentsList();
```
Comment (billyeh) [inline portfolio/src/main/webapp/comments.js]: Do we need the `comment.remove()` and `getCommentsList()`? As I understand it the first call removes the object from the DOM, and the second reloads the info from the database with the comment removed, so are they doing the same thing?

File: portfolio/src/main/webapp/comments.js [hunk 3]
```diff
@@ -46,12 +48,23 @@ async function getCommentsList() {
...
+  });
+  
+  comment.appendChild(deleteButtonElement);
+  list.appendChild(comment);
 }
 
+
```
Comment (billyeh) [inline portfolio/src/main/webapp/comments.js]:
```suggestion
```

File: portfolio/src/main/webapp/comments.js [hunk 4]
```diff
@@ -61,3 +74,11 @@ function submitCommentOnEnter(event) {
     document.getElementById(COMMENT_FORM_ID).submit();
   }
 }
+
+/** Tells the server to delete the comment and refresh the page */
```
Comment (billyeh) [inline portfolio/src/main/webapp/comments.js]:
```suggestion
/** Tells the server to delete the comment and refreshes the page. */
```

File: portfolio/src/main/webapp/comments.js [hunk 5]
```diff
@@ -61,3 +74,11 @@ function submitCommentOnEnter(event) {
...
+async function deleteCommentFromDB(comment) {
+  const params = new URLSearchParams();
+  params.append('id',comment.id);
```
Comment (billyeh) [inline portfolio/src/main/webapp/comments.js]:
```suggestion
  params.append('id', comment.id);
```

File: portfolio/src/main/webapp/comments.js [hunk 6]
```diff
@@ -61,3 +74,11 @@ function submitCommentOnEnter(event) {
...
+  fetch('/delete-comment', {method: 'POST', body: params})
+  .then(getCommentsList());
```
Comment (billyeh) [inline portfolio/src/main/webapp/comments.js]:
```suggestion
    .then(getCommentsList());
```

File: portfolio/src/main/webapp/comments.js (hunk 2 above)
Comment (Obinnabii) [reply portfolio/src/main/webapp/comments.js]: sometimes, there is a lag in the removal and addition of comments.

Comment (Obinnabii) [reply portfolio/src/main/webapp/comments.js]: I don't really know why it happens
```suggestion
```

**Review by billyeh — APPROVED** [review]

File: portfolio/src/main/webapp/comments.js [hunk 7]
```diff
@@ -61,3 +71,11 @@ function submitCommentOnEnter(event) {
     document.getElementById(COMMENT_FORM_ID).submit();
   }
 }
+
+/** Tells the server to delete the comment and refreshes the page. */
+async function deleteCommentFromDB(comment) {
```
Comment (billyeh) [inline portfolio/src/main/webapp/comments.js]:
```suggestion
function deleteCommentFromDB(comment) {
```

**Pull request merged**